app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -74,23 +75,48 @@ def deploy_all
     end
   end
 
+  def create_all_stages_verify
+    @stages_preexisting, @stages_to_create = gather_stages_for_creation
+
+    render
+  end
+
   def create_all_stages
     # No more than one stage, per project, per deploy_group
     # Note: you can call this multiple times, and it will create missing stages, but no redundant stages.
+
+    _throwaway, stages_to_create = gather_stages_for_creation
```

Comment:
`_throwaway` is meaningless ... 

prefer `_, stages = foo` or `stages = foo.last`